code/modules/mechs/equipment/combat.dm

```diff
@@ -802,6 +805,221 @@
 			cocked = FALSE
 			// not being able to fire removes the CH(done in reloadGun now)
 
+/obj/item/mech_equipment/mounted_system/mace
+	name = "\improper NT \"Warhead\" mace"
+	desc = "An exosuit-mounted mace. Doubles as a flail. Handle with care."
+	icon_state = "mech_mace"
+	holding_type = /obj/item/tool/hammer/mace/mech
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	matter = list(MATERIAL_PLASTEEL = 15, MATERIAL_STEEL = 10)
+	origin_tech = list(TECH_COMBAT = 4, TECH_MAGNET = 3)
+	/// Determines what mode our mace is. FALSE is mace, TRUE is flail
+	var/flail_mode = FALSE
+
+/obj/item/mech_equipment/mounted_system/mace/Initialize()
+	. = ..()
+	var/obj/item/tool/hammer/mace/mech/holdin = holding
+	holdin.wielded = TRUE
+
+/obj/item/mech_equipment/mounted_system/mace/attack_self(mob/user)
+	. = ..()
+	if(.)
+		var/obj/item/tool/hammer/mace/mech/holdin = holding
+		holdin.update_flail_mode(flail_mode)
+		flail_mode = !flail_mode
+		owner.update_icon()
+
+/obj/item/mech_equipment/mounted_system/mace/resolve_attackby(mob/living/target, mob/user, params)
+	if(ishuman(target))
+		var/mob/living/carbon/human/targ = target
+		if(targ.stats.getStat(STAT_VIG) > STAT_LEVEL_EXPERT)
+			targ.visible_message(SPAN_DANGER("[targ] dodges the [holding] swing!"), "You dodge [loc]'s [holding] swing!", "You hear a woosh.", 6)
+			return
+	. = ..()
+	if(. && ismech(loc) && istype(target) && target != owner)
+		if(flail_mode)
+			target.visible_message(SPAN_DANGER("[target] gets slammed by [src]'s [holding]!"), SPAN_NOTICE("You get slammed by [src]'s [holding]!"), "You hear something soft hit a metal plate!", 6)
+			target.Weaken(5)
```

Comment:
Weaken is a stun that slows movement to a crawl and prevents fighting back, which we might have problems with if it can be applied repeatedly. A different system like recoil can apply slowdown, while also having a capacity to disarm someone. It might be preferable here.

Reply:
Couldn't find any recoil system. However, I reduced this value from 5 to 3, and the weaken with the mace mode is gone entirely.